src/OpenLoco/src/Main.cpp

```diff
@@ -5,15 +5,45 @@
 #define WIN32_LEAN_AND_MEAN
 #endif
 #include <windows.h>
+
+#include <shellapi.h>
 /**
  * The function that is called directly from the host application (loco.exe)'s WinMain. This will be removed when OpenLoco can
  * be built as a stand alone application.
  */
 // Hack to trick mingw into thinking we forward-declared this function.
 extern "C" __declspec(dllexport) int StartOpenLoco(HINSTANCE hInstance, HINSTANCE hPrevInstance, LPSTR lpCmdLine, int nCmdShow);
-extern "C" __declspec(dllexport) int StartOpenLoco([[maybe_unused]] HINSTANCE hInstance, [[maybe_unused]] HINSTANCE hPrevInstance, LPSTR lpCmdLine, [[maybe_unused]] int nCmdShow)
+extern "C" __declspec(dllexport) int StartOpenLoco([[maybe_unused]] HINSTANCE hInstance, [[maybe_unused]] HINSTANCE hPrevInstance, [[maybe_unused]] LPSTR lpCmdLine, [[maybe_unused]] int nCmdShow)
```

Comment:
Should this still be `extern C`?

Reply:
Yeah we aren't touching the binary entry point.

Reply:
Are you able to keep this logic in the CommandLine class? I say this because I think it is useful to reuse the class in other projects - and converting from the char** argv is an important step.

Reply:
I don't quite see why you would want to have this platform specific code in the command line class. All this is doing is getting the windows side into the same format as non windows side. Okay we could move argc/argv to std::vector into the command line class but then we are just throwing away precomputed lengths and have to mess about to handle the memory.

Reply:
Because `getCmdLineVector` is related to command line, so it seems fitting it to put the command line related code in the command line related class. Even if it is a static method or an alternative constructor.

This then makes it more convenient to lift this class out and use it in another project.

Reply:
`getCmdLineVector` is now just a platform function that puts the arguments into vector of utf8 strings. I can move the call of getCmdLineVector into the command line class but the definition is platform specific so doesn't make sense for it to be in there. If I moved the construction of the std::vector (the non platform bit) into the command line class then we have issues with lifetimes of items and we loose the lengths of the arguments only to immediately recalculate them. A command line class that takes a vector of strings is much easier to test compared to one taking c strings. We would then need to account for null arguments negative arg counts... etc.